src/echo_handler_test.cc

```diff
@@ -0,0 +1,35 @@
+#define TEST_ECHO_HANDLER
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "request.h"
+#include "response.h"
+#include "config_parser.h"
+#include "echo_handler.h"
+
+class EchoHandlerTest : public ::testing::Test {
+ protected:
+  EchoHandlerTest() {
+    status_ = echo_handler.Init(uri_prefix_, config_);
+    request_ = Request::Parse(TEST_BUFFER);
+  }
+
+  EchoHandler echo_handler_;
+  const std::string uri_prefix_;
+  const NginxConfig config_;
```

Comment:
Looks like NginxConfig and uri_prefix_ are never really used later on. Just move it in to the constructor so they're not member variables. (This applies to some of the other files too)